percy/percy-web
# Github Marketplace Purchase

##  [Clubhouse Card](https://app.clubhouse.io/percy/story/1907/require-github-identity-on-github-marketplace-purchases)

## Description
This PR dampens some of the fire of purchasing (and correctly activating) a $99 plan from the github marketplace. Now users that have signed up with a user name and password will be forced to connect their github account before creating an organization.

That way we will properly connect the github purchase to the correct organization for billing purposes.

This is also a bandaid until the signup flow is redesigned.

Want to check it out locally? Pull down this branch and use this link: http://dev.percy.local:4200/organizations/new?marketplace_listing_plan_id=9

## Reviewing this PR
_**Is Covered by Automated Tests?: YESish**_

There are tests in place, but manual testing is required to know that all the parts are working together (`percy-web`, `percy-api`, auth0, and github 😬 ).


### Manual Testing Steps

**note: Don't use the production backend for this. Use `percy-api` locally and make sure it's up to date.**

1. Open up a browser you don't use, or an incognito window. You should be logged out of Percy.
2. You'll need a test account for ~~Team Foundation Server~~ Github. A test account is important because we're going to unlink Percy from that account. It'd be a headache to do it from your main account.

3. Use this url to start the process: http://dev.percy.local:4200/organizations/new?marketplace_listing_plan_id=9 This is the same url github provides when a user makes a marketplace purchase. Remember, you should be signed out of Percy.
4. Setup a new account with Percy using an email and password.
5. Verify your email **in the same browser session** (ie in the same browser or incognito browser window).
6. Log in with your newly created user

### The important part of the manual testing
7. You should now see the org creation page with the connect to github section and the create org section below it, greyed out.
8. Connect your github account.
9. You should be redirected back to the org creation page with the github section greyed out and your github user name displayed with a green check.
10. You should be able to create an org now.


## Pictures
### Before
![screen shot 2018-06-05 at 2 25 02 pm](https://user-images.githubusercontent.com/1929735/40998171-8a862e2e-68cc-11e8-933a-936ad04318f0.png)

### After (GitHub not connected)
<img width="1268" alt="screen shot 2018-06-07 at 2 03 42 pm" src="https://user-images.githubusercontent.com/1929735/41120652-c0c90480-6a5b-11e8-868b-a4ae807f8e66.png">

### After (GitHub Connected)
<img width="1268" alt="screen shot 2018-06-07 at 2 03 50 pm" src="https://user-images.githubusercontent.com/1929735/41120664-cb8a0e5a-6a5b-11e8-9ac5-a109d80987e9.png">

### After (Org Name Entered)
<img width="1268" alt="screen shot 2018-06-07 at 2 04 08 pm" src="https://user-images.githubusercontent.com/1929735/41120678-d86f67f0-6a5b-11e8-87c8-5695cade8f04.png">
